synapse/logging/opentracing.py

```diff
@@ -617,23 +617,25 @@ def set_operation_name(operation_name: str) -> None:
 
 
 @only_if_tracing
-def force_tracing(span=Sentinel) -> None:
+def force_tracing(span: "opentracing.Span" = Sentinel) -> None:  # type: ignore[assignment]
```

Comment:
Elsewhere in synapse we tend to define a dummy enum like so (search for `_Sentinel`):
```py
class _Sentinel(enum.Enum):
    # defining a sentinel in this way allows mypy to correctly handle the
    # type of a dictionary lookup.
    sentinel = object()
```
Then we can avoid the mypy complaints:
```suggestion
def force_tracing(span: Union["opentracing.Span", _Sentinel] = Sentinel) -> None:
```